generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -82,7 +82,8 @@ private boolean mustBeNull(FieldSpec fieldSpec) {
     private Stream<FieldSpec> getNotNullSetRestrictionFilterOnMustContainRestriction(MustContainRestriction restriction) {
         return restriction.getRequiredObjects()
             .stream()
-            .filter(o -> o.getSetRestrictions() != null);
+            .filter(o -> o.getSetRestrictions() != null
```

Comment:
This eliminates the null-pointer issue.

Question is which of the following is true when this happens:
1. The whitelist is null and there is equally no blacklist
1. The whitelist is null and there is a blacklist

If the former is true then the question is why is it in the mustContinsRestriction in the first place.
If the latter is true then the question is should we be using the blacklist, i.e. to eliminate other 'must contain' values?

Reply:
Hi @sl-slaing @pdaulbyscottlogic i think this PR is good to go as it is. Simon raises a very good point as to how we can use the blacklists so I would suggest proceeding with this PR as it is and raising a separate Issue for the suggestion. What do you think? Thanks.